### PR TITLE
feat: 6612 ETL processing docs update

### PR DESCRIPTION
**Pull request by motylp**

### CODE REVIEW

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md:41]: Here I would just reference and link to the new page I added about the example dataset (once merged in).

Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md:41]: OK, I'll wait for yours to get merged first, and maybe will add in there some additional info about this dataset (since I removed it from other places) and then reference it here.

Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: Reference updated, but I left this small description - let me know if you're OK with that.

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (iancowleyamido) [inline docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md:28]: Possibly more descriptive to say:

Conformance, cleansing and data quality processing stage.

File: docs/workloads/azure/data/etl_pipelines/silver_data_azure.md
Comment (iancowleyamido) [inline docs/workloads/azure/data/etl_pipelines/silver_data_azure.md:15]: Probably in this section I might highlight the aims of the silver layer - people may or may not know what it's used for, so talking about cleansing, conformance and data quality is probably good?

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: updated

File: docs/workloads/azure/data/etl_pipelines/silver_data_azure.md
Comment (adurkan-amido) [inline docs/workloads/azure/data/etl_pipelines/silver_data_azure.md:16]:
```suggestion
The Stacks data platform contains an example 'Silver' data processing workload, performing [Bronze to Silver layer](./etl_intro_data_azure.md#data-pipelines) transformations. It accepts raw data of various formats
(e.g. CSV, Parquet, JSON, Delta) as an input and converts it into a more structured Delta format.
```

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:17]:
```suggestion
[PySparkle](https://github.com/amido/stacks-azure-data/tree/main/pysparkle) is a Python library built to streamline
data processing workflows with Apache Spark, and is an integral part of the Stacks Data Platform.
```

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:20]: Following the suggestion above, would delete this.

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:35]: This is repeated elsewhere so think we can remove here.

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:25]: (./etl_intro_data_azure.md#data-pipelines)

```suggestion
Pysparkle is designed to assist implementing a [medallion data lake](./etl_intro_data_azure.md#data-pipelines). The key components of PySparkle include:
```

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:30]:
```suggestion
- **Gold processing**: Capabilities for performing 'Silver to Gold' layer transformations. Supports additional transformations and aggregations of data, creating data sets optimised for downstream consumption.
```

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:28]:
```suggestion
- **Silver processing**: Capabilities for performing 'Bronze to Silver' layer transformations. It accepts raw data of various
formats (e.g. CSV, Parquet, JSON, Delta) as an input and converts it into a more structured Delta format.
```

File: pysparkle/README.md
Comment (adurkan-amido) [inline pysparkle/README.md:1]: Suggestions as for docs/workloads/common/data/pysparkle/pysparkle_quickstart.md

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: fine, done

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: done

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: done

File: docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/etl_intro_data_azure.md]: done

File: docs/workloads/azure/data/etl_pipelines/silver_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/silver_data_azure.md]: done

File: docs/workloads/azure/data/etl_pipelines/silver_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/silver_data_azure.md]: done

File: docs/workloads/azure/data/etl_pipelines/silver_data_azure.md
Comment (motylp) [reply docs/workloads/azure/data/etl_pipelines/silver_data_azure.md]: done, but without the reference link - I think there is no reason to go back to the introduction here.

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

**Review by adurkan-amido — APPROVED** [review]

**Pull request merged**
